internal/node/hdb/types.go

```diff
@@ -20,8 +20,18 @@ type TransitionWrapper struct {
 }
 
 type Transition interface {
+	// Type returns a string constant identifying the type of state transition. Currently these are in snake case.
 	Type() string
+
+	// Patch takes the old state and returns a JSON patch to apply to the old state to get the new state.
 	Patch(oldState []byte) ([]byte, error)
+
+	// Enrich adds important data to the transition that is not specified when submitted by the client.
+	// For example, the id of a newly created object can be generated during the enrichment phase. The id is not
+	// something we'd want the client to submit, but we want it to be included in the transition.
+	Enrich(oldState []byte) error
```

Comment:
can we make `oldState []byte` -> `oldState State` and have `type State []byte`? that way we can change out the type underneath later if we need.

Also, why can't we move this itno `core/state/node/transitions`? it feels weird that this interface type is in `internal/node` when it's actually used in `core`

Reply:
I think a lot of the weirdess with types is that I'm trying to keep the state machine as generic as possible since we will eventually need it to handle different schemas (I'm thinking about community schemas down the line). But yeah a type alias here will help with readability. I'm thinking `SerializedState` because we already have `State` reserved.